backend/courses/models.py

```diff
@@ -1419,3 +1419,40 @@ def create_or_update_user_profile(sender, instance, created, **kwargs):
     if created and instance.email != "":
         instance.profile.email = instance.email
     instance.profile.save()
+
+
+
+class Friendship(models.Model):
+    """
+    Used to track friendships along with requests status 
+    """
+    
+    requestSender = models.ForeignKey(
+        UserProfile,
+        on_delete=models.CASCADE,
+        related_name="sender",
+        help_text="The person (user) who sent the request.",
+    )
+
+    requestReciever = models.ForeignKey(
+        UserProfile,
+        related_name="receiver",
+         on_delete=models.CASCADE,
+        help_text="The person (user) who recieved the request.",
+    )
+
+    status = models.IntegerField(
```

Comment:
Could we replace this with an enum of `SENT`, `ACCEPTED`, and `REJECTED` ? Django has built-in support for enums: https://stackoverflow.com/a/58051918